### PR TITLE
macOS 12 changes

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
# Description
In scope of this pull request we add preview support for macOS 12 Monterey.
**For new tools, please provide total size and installation time.**

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 1]
```diff
@@ -77,13 +82,21 @@ $packageManagementList = @(
     (Get-HomebrewVersion),
     (Get-NPMVersion),
     (Get-YarnVersion),
-    (Get-NuGetVersion),
-    (Get-CondaVersion),
+    (Get-NuGetVersion)
+)
+
+if (-not $os.IsMonterey) {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Generator.ps1:88]: `$os.IsLessThanMonterey`

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 2]
```diff
@@ -77,13 +82,21 @@ $packageManagementList = @(
...
+    )
+}
+
+$packageManagementList += @(
     (Get-RubyGemsVersion),
     (Get-ComposerVersion)
 )
 
-if ($os.IsHigherThanMojave) {
+if ($os.IsHigherThanMojave -and -not $os.IsMonterey) {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Generator.ps1:99]: `$os.IsLessThanMonterey`

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 1 above)
Comment (dmitry-shibanov) [reply images/macos/software-report/SoftwareReport.Generator.ps1]: Done. Thank you.

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 2 above)
Comment (dmitry-shibanov) [reply images/macos/software-report/SoftwareReport.Generator.ps1]: Done. Thank you.

File: images/macos/helpers/Common.Helpers.psm1 [hunk 3]
```diff
@@ -28,16 +28,20 @@ function Get-EnvironmentVariable($variable) {
 function Get-OSVersion {
     $osVersion = [Environment]::OSVersion
     $osVersionMajorMinor = $osVersion.Version.ToString(2)
+    $osMontereyVersion = [SemVer]$(sw_vers -productVersion)
```
Comment (maxim-lobanov) [inline images/macos/helpers/Common.Helpers.psm1:31]: put the comment why we need the separate command

File: images/macos/provision/core/aws.sh [hunk 4]
```diff
@@ -6,9 +6,11 @@ AWS_CLI_URL="https://awscli.amazonaws.com/AWSCLIV2.pkg"
 download_with_retries $AWS_CLI_URL "/tmp"
 sudo installer -pkg /tmp/AWSCLIV2.pkg -target /
 
-echo Installing aws sam cli...
-brew tap aws/tap
-brew install aws-sam-cli
+if is_Less_Monterey; then
```
Comment (maxim-lobanov) [inline images/macos/provision/core/aws.sh:9]: Put the comment why we don't install it on Monterey for now

File: images/macos/provision/core/homebrew.sh [hunk 5]
```diff
@@ -13,8 +13,10 @@ brew analytics off
 echo "Installing jq..."
 brew install jq
 
-echo "Installing curl..."
-brew_smart_install "curl"
+if is_Less_Monterey; then
```
Comment (maxim-lobanov) [inline images/macos/provision/core/homebrew.sh:16]: Put the comment why we don't install it on Monterey for now

File: images/macos/provision/core/node.sh [hunk 6]
```diff
@@ -29,10 +29,12 @@ else
   brew_smart_install "node@14"
   brew link node@14 --force
 
-  for module in ${node_modules[@]}; do
-    echo "Install $module"
-    npm install -g $module
-  done
+  if is_Less_Monterey; then
```
Comment (maxim-lobanov) [inline images/macos/provision/core/node.sh:32]: Put the comment why we don't install it on Monterey for now

File: images/macos/provision/core/python.sh [hunk 7]
```diff
@@ -3,14 +3,16 @@ source ~/utils/utils.sh
...
 
-echo "Install latest Python 2"
-Python2Url="https://www.python.org/ftp/python/2.7.18/python-2.7.18-macosx10.9.pkg"
-download_with_retries $Python2Url "/tmp" "python2.pkg"
-sudo installer -pkg /tmp/python2.pkg -target /
-pip install --upgrade pip
-
-echo "Install Python2 certificates"
-bash -c "/Applications/Python\ 2.7/Install\ Certificates.command"
+if is_Less_Monterey; then
```
Comment (maxim-lobanov) [inline images/macos/provision/core/python.sh:6]: Put the comment why we don't install it on Monterey for now

File: images/macos/provision/core/ruby.sh [hunk 8]
```diff
@@ -2,7 +2,15 @@
 source ~/utils/utils.sh
 
 echo Installing Ruby...
-brew_smart_install "ruby@2.7"
+
+if is_Monterey; then
```
Comment (maxim-lobanov) [inline images/macos/provision/core/ruby.sh:6]: Put the comment why we don't install it on Monterey for now

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 9]
```diff
@@ -49,10 +49,15 @@ $languageAndRuntimeList = @(
 
 if ( -not $os.IsHighSierra) {
     $languageAndRuntimeList += @(
-        (Get-GccVersion)
         (Get-FortranVersion)
-        (Get-ClangLLVMVersion)
     )
+
+    if ( -not $os.IsMonterey) {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Generator.ps1:55]: IsLessThanMonterey

**Pull request closed without merging**